libmscore/cmd.cpp

```diff
@@ -2693,12 +2694,16 @@ void Score::cmdImplode()
             // identify tracks to combine, storing the source track numbers in tracks[]
             // first four non-empty tracks to win
             for (int track = startTrack; track < endTrack && full < VOICES; ++track) {
-                  for (Measure* m = startMeasure; m && m != endMeasure; m = m->nextMeasure()) {
+                  Measure* m = startMeasure;
+                  do {
                         if (m->hasVoice(track) && !m->isOnlyRests(track)) {
                               tracks[full++] = track;
                               break;
                               }
-                        }
+                        if (m == endMeasure)
+                              break;
+                        m = m->nextMeasure();
+                        } while (m);
```

Comment:
I could combine these 4 lines into just one:
```
} while ((m != endMeasure) && (m = m->nextMeasure()))
```

Reply:
Combine, please. This is a common notation involving pointers and must be clear to a reader.